### PR TITLE
Check for full path for log-bin to stop puppet from managing directory “."

### PR DESCRIPTION
**Pull request by NoodlesNZ**
As reported by @larkit-ian in PR #596

### CODE REVIEW

File: manifests/server/config.pp [hunk 1]
```diff
@@ -23,11 +23,15 @@
...
   if $logbin {
     $logbindir = dirname($logbin)
-    file { $logbindir:
-      ensure => directory,
-      mode   => '0755',
-      owner  => $options['mysqld']['user'],
-      group  => $options['mysqld']['user'],
+    
+    #Stop puppet from managing directory if just a filename/prefix is specified
+    if $logbindir != '.' {
```
Comment (igalic) [inline manifests/server/config.pp]: i'm confused about this: when do we (anyone?) set logbindir to .? how do we know to set it to .?

**Comment by larkit-ian** [comment]
As I read it, if we don't specify an absolute path, then dirname($logbin) will set $logbindir to ".".  In that case, we don't to have the file resource for that dir.

**Comment by JCotton1123** [comment]
I am also currently running into this issue. We can please get this merged ASAP?

@NoodlesNZ it looks like you need to rebase your branch

**Comment by NoodlesNZ** [comment]
I have rebased, but CI is taking a while.

**Comment by igalic** [comment]
i see now how this makes sense ~_~

**Comment by NoodlesNZ** [comment]
:) Thanks for the merge

File: manifests/server/config.pp [hunk 2]
```diff
@@ -22,12 +22,16 @@
   $logbin = pick($options['mysqld']['log-bin'], $options['mysqld']['log_bin'], false)
 
   if $logbin {
-    $logbindir = mysql_dirname($logbin)
-    file { $logbindir:
-      ensure => directory,
-      mode   => '0755',
-      owner  => $options['mysqld']['user'],
-      group  => $options['mysqld']['user'],
+    $logbindir = dirname($logbin)
```
Comment (mhaskel) [inline manifests/server/config.pp]: This needs to use `mysql_dirname` not `dirname` as the stdlib requirement is > 3.2.0.

Comment (mhaskel) [inline manifests/server/config.pp]: Issued #677 to fix this.

**Pull request merged**